packages/auth-construct/src/construct.ts

```diff
@@ -900,132 +902,209 @@ export class AmplifyAuth
+      produce: () => {
+        if (cfnUserPool.policies) {
+          const policy = (cfnUserPool.policies as CfnUserPool.PoliciesProperty)
+            .passwordPolicy as CfnUserPool.PasswordPolicyProperty;
+          return policy.minimumLength?.toString();
+        }
+        return undefined;
+      },
+    });
 
-      const requirements = [];
-      if (this.computedUserPoolProps.passwordPolicy.requireDigits) {
-        requirements.push('REQUIRES_NUMBERS');
-      }
-      if (this.computedUserPoolProps.passwordPolicy.requireLowercase) {
-        requirements.push('REQUIRES_LOWERCASE');
-      }
-      if (this.computedUserPoolProps.passwordPolicy.requireUppercase) {
-        requirements.push('REQUIRES_UPPERCASE');
-      }
-      if (this.computedUserPoolProps.passwordPolicy.requireSymbols) {
-        requirements.push('REQUIRES_SYMBOLS');
-      }
+    output.passwordPolicyRequirements = Lazy.string({
+      produce: () => {
+        if (cfnUserPool.policies) {
+          const policy = (cfnUserPool.policies as CfnUserPool.PoliciesProperty)
+            .passwordPolicy as CfnUserPool.PasswordPolicyProperty;
+          const requirements = [];
+          policy.requireNumbers && requirements.push('REQUIRES_NUMBERS');
+          policy.requireLowercase && requirements.push('REQUIRES_LOWERCASE');
+          policy.requireUppercase && requirements.push('REQUIRES_UPPERCASE');
+          policy.requireSymbols && requirements.push('REQUIRES_SYMBOLS');
+          return JSON.stringify(requirements);
+        }
+        return undefined;
+      },
+    });
 
-      if (requirements.length > 0) {
-        output.passwordPolicyRequirements = JSON.stringify(requirements);
-      }
-    }
+    // extract the MFA configuration setting from the UserPool resource
+    output.mfaConfiguration = Lazy.string({
+      produce: () => {
+        return cfnUserPool.mfaConfiguration ?? 'OFF';
+      },
+    });
+    // extract the MFA types from the UserPool resource
+    output.mfaTypes = Lazy.string({
+      produce: () => {
+        const mfaTypes: string[] = [];
+        (cfnUserPool.enabledMfas ?? []).forEach((type) => {
+          if (type === 'SMS_MFA') {
+            mfaTypes.push('SMS');
+          }
+          if (type === 'SOFTWARE_TOKEN_MFA') {
+            mfaTypes.push('TOTP');
+          }
+        });
+        return JSON.stringify(mfaTypes);
+      },
+    });
 
-    if (this.computedUserPoolProps.mfa) {
-      output.mfaConfiguration = this.computedUserPoolProps.mfa;
+    // extract social providers from UserPool resource
+    output.socialProviders = Lazy.string({
+      produce: () => {
+        const outputProviders = [];
+        const userPoolProviders = this.resources.userPool.identityProviders;
+        if (userPoolProviders) {
+          for (const provider of userPoolProviders) {
+            const providerType =
+              provider.node['_children']['Resource']['providerType'];
```

Comment:
Pls use dot notations in this file as well

Reply:
updated to use findResource() combined with casting; that helped eliminate all usage of '_children' etc